### PR TITLE
Added environment creation and doc tools install

### LINKED ISSUE #46: Add environment setup steps to contributing guide
pointed out [in zulip](https://napari.zulipchat.com/#narrow/stream/246864-sprints/topic/Help/near/203604336), our readme emphasises creating a new environment when working with napari, but the contributing guide `docs/source/developers/contributing` does not.  Could create issues for a new developer trying to work on napari in an old or otherwise cluttered environment

### PR DESCRIPTION
**Pull request by melissawm**
# Description

This is a redo of napari/napari#4827. Also includes some reorganization on the contributing guide. cc @rwkozar

Co-authored-by: Rob Kozar <rwkozar88@yahoo.com>

## Type of change

- [x] Fixes or improves existing content
- [ ] Adds new content page(s)
- [ ] Fixes or improves workflow, documentation build or deployment

# References
Closes #46 
Depends on #63 for the nice sphinx-design tabs :)

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added [alt text](https://webaim.org/techniques/alttext/) to new images included in this PR

### CODE REVIEW

File: docs/developers/contributing.md [hunk 1]
```diff
@@ -9,55 +9,95 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    After installing Python on your machine, create a virtual environment on your terminal and activate it. On Linux and MacOS, you can run
+    ```sh
+    python -m venv <path-to-env>
+    source <path-to-env>/bin/activate
+    ```
+    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
+    :::
+
+    :::{tab-item} Using `conda`
+    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` with Python 3.9 and activate it.
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env`—here with Python 3.9—and activate it.
```

File: docs/developers/contributing.md [hunk 2]
```diff
@@ -9,55 +9,95 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    development environment.
+    ```
+
+4. Install the package in editable mode, along with all of the developer tools.
+
+    If you only want to use napari, you can install it on most macOS, Linux and
+    Windows systems with
+    ![Supported Python versions](https://img.shields.io/pypi/pyversions/napari.svg)
+    by following the directions on the
+    [instructions page](installation#install-as-python-package-recommended).
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).
```
For me the current link is missing, I think this should be correct

Comment (psobolewskiPhD) [reply docs/developers/contributing.md]: Not sure I wouldn't just remove this paragraph about the standard installation anyways.

File: docs/developers/contributing.md [hunk 3]
```diff
@@ -9,55 +9,95 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    To install the package in editable mode for development, along with all of
+    the developer tools, use the following command:
+    ```sh
+    pip install -e ".[dev]"  # (quotes only needed for zsh shell)
+    ```
+
+5. We use [`pre-commit`](https://pre-commit.com) to sort imports with
+   [`isort`](https://github.com/PyCQA/isort), format code with
+   [`black`](https://github.com/psf/black), and lint with
+   [`flake8`](https://github.com/PyCQA/flake8) automatically prior to each commit.
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
5. We use [`pre-commit`](https://pre-commit.com) to identify and fix some style issues automatically prior to each commit. Currently, this includes sorting imports with
   [`isort`](https://github.com/PyCQA/isort), formating code with
   [`black`](https://github.com/psf/black), and linting with
   [`flake8`](https://github.com/PyCQA/flake8).
```

File: docs/developers/contributing.md [hunk 4]
```diff
@@ -209,17 +249,9 @@ git push -u origin your-branch-name
...
-```sh
-make docs
-```
-
-The docs will be built at `docs/_build/html`.
+## Contributing documentation
 
-Most web browsers will allow you to preview HTML pages.
-Try entering `file:///absolute/path/to/napari/docs/_build/html/index.html` in your address bar.
+If you wish to contribute documentation changes to napari, please read the [guide on contributing documentation](documentation/index.md).
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
If you wish to contribute to the napari documentation, please read the [guide on contributing documentation](documentation/index.md). 
```

File: docs/developers/contributing.md [hunk 5]
```diff
@@ -9,55 +9,95 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    It is highly recommended to create a fresh environment when working with
+    napari, to prevent issues with outdated or conflicting packages in your
+    development environment.
+    ```
+
+4. Install the package in editable mode, along with all of the developer tools.
+
+    If you only want to use napari, you can install it on most macOS, Linux and
+    Windows systems with
+    ![Supported Python versions](https://img.shields.io/pypi/pyversions/napari.svg)
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]: I personally don't like the look of the shields in-line with text:
image

But it's not a blocker for me.

File: docs/index.md [hunk 6]
```diff
@@ -94,9 +94,9 @@ you can install via pip, conda-forge, or from source.
 
 ### From pip, with "batteries included"
 
-napari can be installed on most macOS, Linux, and Windows systems with Python
-3.8-3.10 using pip:
-
+napari can be installed on most macOS, Linux, and Windows systems with
+ ![Supported Python versions](https://img.shields.io/pypi/pyversions/napari.svg) by
```
Comment (psobolewskiPhD) [inline docs/index.md]: I personally don't like the shields in-line with text, but it's not a blocker.
image

**Review by psobolewskiPhD — DISMISSED** [review]
Made some minor changes.
There's some changes in the Installation too including the python version. That is also being changed here https://github.com/napari/docs/pull/51 with https://github.com/Chris-N-K/napari-docs/pull/2
to be able to change python version in conf.py and use substitutions.

BTW I'm not sure we're officially recommending 3.10 now? CC: @jni

**Comment by melissawm** [comment]
Thanks @psobolewskiPhD - I think I addressed most of your suggestions. The shields had been proposed by the original author - I am ok either way.

**Comment by psobolewskiPhD** [comment]
@melissawm Instead of the shields, perhaps this is a good place to use the version substitution  https://github.com/napari/docs/pull/84

This could be merged as is and then editted in that PR or the other way around. 🤷‍♂️

**Comment by psobolewskiPhD** [comment]
Actually, the conda env creation could also be replaced with the substitution from https://github.com/napari/docs/pull/84

**Comment by melissawm** [comment]
That's a good idea - let me do that!

**Comment by psobolewskiPhD** [comment]
@melissawm So maybe we should get #84 merged before this one?

**Comment by melissawm** [comment]
Yes - that would be helpful!

**Comment by psobolewskiPhD** [comment]
> Yes - that would be helpful!

Merged!

File: docs/developers/contributing.md [hunk 7]
```diff
@@ -9,55 +9,96 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    After installing Python on your machine, create a virtual environment on your terminal and activate it. On Linux and MacOS, you can run
+    ```sh
+    python -m venv <path-to-env>
+    source <path-to-env>/bin/activate
+    ```
+    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
+    :::
+
+    :::{tab-item} Using `conda`
+    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` - here with Python 3.9 - and activate it.
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
    After [installing `conda`](https://www.anaconda.com/products/distribution), create an environment called `napari-env` - here with Python {{ python_version }} - and activate it.
```

File: docs/developers/contributing.md [hunk 8]
```diff
@@ -9,55 +9,96 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    ```sh
+    conda create -y --name napari-env -c conda-forge python=3.9
+    conda activate napari-env
+    ```
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
    {{ conda_create_env }}
```

File: docs/developers/contributing.md [hunk 9]
```diff
@@ -9,55 +9,96 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    It is highly recommended to create a fresh environment when working with
+    napari, to prevent issues with outdated or conflicting packages in your
+    development environment.
+    ```
+
+4. Install the package in editable mode, along with all of the developer tools.
+
+    If you only want to use napari, you can install it on most macOS, Linux and
+    Windows systems with
+    ![Supported Python versions](https://img.shields.io/pypi/pyversions/napari.svg)
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
    Python {{ python_version_range }}
```

File: docs/developers/contributing.md [hunk 10]
```diff
@@ -9,55 +9,96 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    by following the directions on the
+    [instructions page](../tutorials/fundamentals/installation.md#install-as-python-package-recommended).
+
+    To install the package in editable mode for development, along with all of
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md]:
```suggestion
    To install the package in editable mode for development, along with PyQt5 and all of
```

File: docs/developers/contributing.md [hunk 11]
```diff
@@ -9,55 +9,96 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    the developer tools, use the following command:
+    ```sh
+    pip install -e ".[dev]"  # (quotes only needed for zsh shell)
```
Comment (psobolewskiPhD) [inline docs/developers/contributing.md:75]:
```suggestion
    pip install -e ".[pyqt,dev]"  # (quotes only needed for zsh shell)
```

File: docs/index.md (hunk 6 above)
Comment (psobolewskiPhD) [inline docs/index.md]:
```suggestion
Python {{ python_version_range }}
```

File: docs/index.md [hunk 12]
```diff
@@ -110,7 +110,7 @@ napari into a clean virtual environment using an environment manager like
 [venv](https://docs.python.org/3/library/venv.html).  For example, with `conda`:
 
 ```sh
-conda create -y -n napari-env -c conda-forge python=3.9
+conda create -y -n napari-env -c conda-forge python=3.10
```
Comment (psobolewskiPhD) [inline docs/index.md]: I asked core-devs and for now we prefer to recommend 3.9.
This line will be a merge conflict, so keep what was merged with the conda_subs

File: docs/tutorials/fundamentals/installation.md [hunk 13]
```diff
@@ -68,7 +68,7 @@ napari into a clean virtual environment using an environment manager like
 This should be set up *before* you install napari. For example, with `conda`:
 
 ```sh
-conda create -y -n napari-env -c conda-forge python=3.9
+conda create -y -n napari-env -c conda-forge python=3.10
```
Comment (psobolewskiPhD) [inline docs/tutorials/fundamentals/installation.md]: I asked core-devs and for now we prefer to recommend 3.9.
This line will be a merge conflict, so keep what was merged with the conda_subs

**Comment by psobolewskiPhD** [comment]
I made suggestions for using the #84 substitutions.
Also I asked core-devs and for now we prefer to recommend python 3.9.
So in the conda envs, the existing (#84 merged) version is preferred.

**Comment by melissawm** [comment]
I think I got all your comments - let me know if there's anything else. Cheers!

**Review by psobolewskiPhD — APPROVED** [review]

File: docs/developers/contributing.md [hunk 14]
```diff
@@ -9,69 +9,103 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    :::{tab-item} Using `venv`
+    After installing Python on your machine, create a virtual environment on your terminal and activate it. On Linux and MacOS, you can run
+    ```sh
+    python -m venv <path-to-env>
+    source <path-to-env>/bin/activate
+    ```
+    See the [venv](https://docs.python.org/3/library/venv.html) documentation for instructions on Windows.
+    :::
 
+    :::{tab-item} Using `conda`
```
Comment (DragaDoncila) [inline docs/developers/contributing.md]: I wonder if we should put the `conda` instructions first? I think it's far more commonly used, and then for stuff like the arm64 mac stuff you basically have to use `conda` if I understand correctly?

File: docs/developers/contributing.md [hunk 15]
```diff
@@ -9,69 +9,103 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    ```sh
+    pip install -e ".[pyqt,dev]"  # (quotes only needed for zsh shell)
+    ```
+
+    If you want to use PySide2 instead, you would use:
+    ```sh
+    pip install -e ".[pyside,dev]"  # (quotes only needed for zsh shell)
+    ```
+
+    Finally, if you already have a Qt backend installed or want to use an experimental one like Qt6 use:
```
Comment (DragaDoncila) [inline docs/developers/contributing.md:73]: Would be good to add a small note or comment here that you will then also need to manually install your backend. I know it's obvious maybe but still

File: docs/developers/contributing.md [hunk 16]
```diff
@@ -223,17 +257,9 @@ git push -u origin your-branch-name
...
 
-## Building the docs
-
-From the project root:
-```sh
-make docs
-```
-
-The docs will be built at `docs/_build/html`.
+## Contributing documentation
```
Comment (DragaDoncila) [inline docs/developers/contributing.md]: I would actually move this section up to just underneath setting up a `dev` installation. Pretty much everything else on this page is less important and less frequently used imo.

**Review by DragaDoncila — CHANGES_REQUESTED** [review]
Hi @melissawm thank you for these updates! My apologies for the delayed review as well - I've made just three small comments, but it's looking good!

File: docs/developers/contributing.md (hunk 15 above)
Comment (melissawm) [reply docs/developers/contributing.md:73]: Let me know if this is enough

File: docs/developers/contributing.md (hunk 14 above)
Comment (melissawm) [reply docs/developers/contributing.md]: Done

File: docs/developers/contributing.md (hunk 16 above)
Comment (melissawm) [reply docs/developers/contributing.md]: Moved, thanks!

File: docs/developers/contributing.md [hunk 17]
```diff
@@ -9,69 +9,109 @@ If you are a new user, we recommend checking out the detailed [Github Docs](http
...
+    ```sh
+    pip install -e ".[dev]"  # (quotes only needed for zsh shell)
+    ```
+
+    Note that in this last case you will need to install your Qt backend separately.
+
+5. We use [`pre-commit`](https://pre-commit.com) to sort imports with
+   [`isort`](https://github.com/PyCQA/isort), format code with
+   [`black`](https://github.com/psf/black), and lint with
+   [`flake8`](https://github.com/PyCQA/flake8) automatically prior to each commit.
```
Comment (Czaki) [inline docs/developers/contributing.md]:
```suggestion
5. We use [`pre-commit`](https://pre-commit.com) to format code with
   [`black`](https://github.com/psf/black) and lint with
   [`ruff`](https://github.com/charliermarsh/ruff) automatically prior to each commit.
```

Comment (psobolewskiPhD) [reply docs/developers/contributing.md]: @Czaki 
Should be:
We use pre-commit `to` format code with black`<delete COMMA>` and lint with ...`

Comment (Czaki) [reply docs/developers/contributing.md]: @psobolewskiPhD fixed

**Review by DragaDoncila — APPROVED** [review]
Looks good to me! 🎉

**Comment by melissawm** [comment]
@psobolewskiPhD conflicts are fixed :)

**Pull request merged**